src/core/lib/security/authorization/audit_logging.cc

```diff
@@ -39,62 +40,65 @@
 namespace grpc_core {
 namespace experimental {
 
-void AuditLoggerRegistry::RegisterAuditLoggerFactory(
+namespace {
+
+Mutex* g_mu = new Mutex();
+AuditLoggerRegistry* g_registry ABSL_GUARDED_BY(g_mu) =
+    new AuditLoggerRegistry();
+
+}  // namespace
+
+void AuditLoggerRegistry::RegisterFactory(
     std::unique_ptr<AuditLoggerFactory> factory) {
-  auto& registry = GetAuditLoggerRegistry();
-  MutexLock lock(&registry.mu_);
-  registry.logger_factories_map_[factory->name()] = std::move(factory);
+  if (factory == nullptr) return;
+  MutexLock lock(g_mu);
+  GPR_ASSERT(g_registry->logger_factories_map_.find(factory->name()) ==
```

Comment:
Suggest writing this as follows, to avoid doing the lookup twice:

```
absl::string_view name = factory->name();
GPR_ASSERT(
    g_registry->logger_factories_map_.emplace(name, std::move(factory))
        .second);
```

Reply:
Done.